app/backend/lib/emails/handleEmailNotification.ts

```diff
@@ -0,0 +1,99 @@
+import getConfig from 'next/config';
+import getAuthRole from '../../../utils/getAuthRole';
+import getAuthUser from '../../../utils/getAuthUser';
+import getAccessToken from '../ches/getAccessToken';
+import { performQuery } from '../graphql';
+import config from '../../../config';
+import sendEmail from '../ches/sendEmail';
+
+const getAnalystEmailsByIds = `
+  query getAnalystEmailsByIds($_rowIds: [Int!]!) {
+    allAnalysts(filter: {rowId: {in: $_rowIds}}) {
+      edges {
+        node {
+          email
+        }
+      }
+    }
+  }
+`;
+
+export interface EmailTemplate {
+  emailTo: number[];
+  tag: string;
+  emailCC: number[];
+  subject: any;
+  body: any;
+}
+
+const isAuthorized = (authRole: any) => {
+  const authorizedRoles = ['ccbc_admin', 'ccbc_analyst'];
+  return authRole && authorizedRoles.includes(authRole.pgRole);
+};
+
+/**
+ * Temporary function to get email recipients from the database based on hardcoded IDs
+ * To be replaced with proper email notification system pulls from DB
+ */
+const getEmailRecipients = async (ids: number[], req: any) => {
+  const publicRuntimeConfig = getConfig()?.publicRuntimeConfig;
+  const namespace = publicRuntimeConfig?.OPENSHIFT_APP_NAMESPACE;
+  const isProd = namespace?.endsWith('-prod');
+  if (!isProd || !ids || ids.length === 0) {
+    return [config.get('CHES_TO_EMAIL')];
+  }
+
+  const results = await performQuery(
+    getAnalystEmailsByIds,
+    { _rowIds: ids },
+    req
+  );
+  return results?.data?.allAnalysts?.edges.map((edge) => edge.node.email);
+};
+
+const handleEmailNotification = async (
+  req,
+  res,
+  template: (
```

Comment:
I think making this function an interface here so it can be extended for the template functions with some descriptions of the parameter is a good idea. Also eventInitiator and params should be optional parameters so we don't get type errors.

Reply:
Extracted to an `EmailTemplateProvider` and marked optional for eventInitiator and params